subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StrictDependenciesIntegrationTest.groovy

```diff
@@ -67,16 +61,26 @@ class StrictDependenciesIntegrationTest extends AbstractHttpDependencyResolution
 
     void "should fail if transitive dependency version is not compatible with the strict dependency version"() {
         given:
-        def foo10 = mavenHttpRepo.module("org", "foo", '1.0').publish()
-        def foo11 = mavenHttpRepo.module("org", "foo", '1.1').publish()
-        def bar10 = mavenHttpRepo.module('org', 'bar', '1.0')
-            .dependsOn(foo11)
-            .publish()
+        repository {
+            'org' {
+                'foo' {
+                    '1.0' {
+                        expectGetMetadata()
+                    }
+                    '1.1'()
+                }
+                'bar' {
+                    '1.0' {
+                        dependsOn('org:foo:1.1')
+                        expectGetMetadata()
+                    }
+                }
+            }
+        }
```

Comment:
This reads better, but is less amenable to a fluent-api equivalent, and seems a bit more 'magic'.
Would it be possible to add:
```
repository {
    'org:foo' {
         '1.0' ()
         '1.1' ()
    }
    'org:bar:1.0' ()
}
```

Reply:
It's already there :)